### ISSUE TITLE
Non-blocking KeyError:  'repeat' when Handling Keyboard Event in NiceGUI

### ISSUE THREAD

**Issue by Neu-bee** [zauberzeug/nicegui#2545]
### Description

Hello NiceGUI Community,

I've encountered a `KeyError` when trying to run my NiceGUI application, specifically when handling a keyboard event. The error message indicates that the key `'repeat'` is missing from the event arguments dictionary. I would appreciate any insights or suggestions on how to resolve this issue.

Error Message:
``` cmd
KeyError: 'repeat'
ERROR:nicegui:'repeat'
Traceback (most recent call last):
  File "D:\DevApps\Anaconda\envs\python10Study\lib\site-packages\nicegui\events.py", line 400, in handle_event
    result = handler(arguments) if expects_arguments else handler()
  File "D:\DevApps\Anaconda\envs\python10Study\lib\site-packages\nicegui\elements\keyboard.py", line 43, in _handle_key
    repeat=e.args['repeat'],
KeyError: 'repeat'
``` 

The error occurs when the `_handle_key` method in `elements/keyboard.py` is invoked, and the application attempts to access the `'repeat'` key from the `e.args` dictionary, which seems to be absent.

I am currently using NiceGUI version 1.4.14. 
Below is the code snippet where the error occurs :

``` python
def make_dialog() -> ui.dialog:
    passwords = {'user1': 'pass1', 'user2': 'pass2'}
    with ui.dialog() as dialog, ui.card():
        def try_login() -> None:  # local function to avoid passing username and password as arguments
            if passwords.get(username.value) == password.value:
                app.storage.user.update({'username': username.value, 'authenticated': True})
                log_logic.refresh()
                dialog.close()
            else:
                ui.notify('Wrong username or password', color='negative')

        with dialog, ui.card().classes('w-100 h-100 justify-start items-start absolute-center'):
            username = ui.input('Username').on('keydown.enter', try_login)
            password = ui.input('Password', password=True, password_toggle_button=True).on('keydown.enter', try_login)
            ui.button('Log in', on_click=try_login)
    return dialog


@ui.refreshable
def log_logic():
    with ui.element().classes('max-[420px]:hidden'):
        log_btn = ui.button('Login', on_click=lambda: make_dialog().open()).props('flat color=white round')
        log_out_btn = ui.button('user').props('flat color=white round')
        with log_out_btn:
            with ui.menu().classes('bg-primary text-white text-lg'):
                ui.menu_item("Exit", on_click=lambda: (app.storage.user.clear(), log_logic.refresh()))

        log_out_btn.visible = app.storage.user.get('authenticated', False)
        log_btn.visible = not app.storage.user.get('authenticated', False)

log_logic()
```

The application's functionality is not impacted by this error. 
Thank you for your time and help!

**Comment by falkoschindler** [comment]
Hi @Neu-bee, thanks for reporting this issue.

I'm still struggling to understand the problem:

- Your code doesn't run without a page function, so I added one.
- Then there is nothing to see, unless you run it in dark mode, so I did.
- The you need to make sure to use a screen width larger than 420px, otherwise the buttons are hidden.
- But even then I don't know how to reproduce the KeyError.

So can you, please, reduce the code to a minimum example, make sure it runs, and explain briefly the steps it takes to reproduce the problem? Thanks a lot!

**Comment by Neu-bee** [comment]
Hello again,

I would like to apologize for any confusion caused by my initial report. Upon further investigation, I've realized that the `KeyError: 'repeat'` was a result of a conflict between the 'keydown.enter' event in a login input field and the default enter key behavior in the search bar, both of which I had placed in the same header.

I've also found a new issue that seems to be with the official search bar on the NiceGUI website:

1. When you first load the page and press enter, the search does not activate.
2. However, if you click on the search bar, press ESC to exit, and then press enter again, the search bar pops up.

It appears there's an inconsistency with how the search bar handles the enter key press, particularly after a page refresh followed by the ESC key usage.

Thank you for your assistance and patience.

Best regards!
